### PR TITLE
fix: remove Select key value selects, fix default

### PR DESCRIPTION
**Pull request by znamenskii-ilia**
## Description
https://theappsmith.slack.com/archives/C025SE88KNE/p1725434811592439

Fixes #`Issue Number`  
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/10772310432>
> Commit: 3afce8dcc618f3318cfff59fa3940f398091cdd1
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=10772310432&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.All`
> Spec:
> <hr>Mon, 09 Sep 2024 12:43:13 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Introduced enhanced validation for options in the WDSComboBoxWidget, ensuring all entries are valid objects with `label` and `value` properties.
	- Updated WDSSelectWidget to include validation for the options array, ensuring only valid items are processed.

- **Bug Fixes**
	- Removed outdated validation methods and configurations in WDSSelectWidget for a more streamlined approach.

- **Refactor**
	- Simplified the configuration and validation logic within the WDSSelectWidget, improving maintainability and clarity.

### CODE REVIEW

**Review by jsartisan — APPROVED** [review]

File: app/client/src/widgets/wds/WDSComboBoxWidget/config/propertyPaneConfig/validations/optionsCustomValidation.ts [hunk 1]
```diff
@@ -77,6 +77,22 @@ export function optionsCustomValidation(
...
 
+    const isValidKeys = options.every((option) => {
+      return (
+        _.isPlainObject(option) &&
+        _.has(option, "label") &&
+        _.has(option, "value")
+      );
+    });
+
+    if (!isValidKeys) {
```
Comment (sagar-qa007) [inline app/client/src/widgets/wds/WDSComboBoxWidget/config/propertyPaneConfig/validations/optionsCustomValidation.ts:88]: Should we consider updating existing test cases or adding new cases for Unit/Cypress?
Your insights would be appreciated.

**Review by sagar-qa007 — DISMISSED** [review]

File: app/client/src/widgets/wds/WDSComboBoxWidget/config/propertyPaneConfig/validations/optionsCustomValidation.ts (hunk 1 above)
Comment (znamenskii-ilia) [reply app/client/src/widgets/wds/WDSComboBoxWidget/config/propertyPaneConfig/validations/optionsCustomValidation.ts:88]: I'm going to skip it for now and return to it later.

**Pull request merged**